### ISSUE TITLE
Is the provided binary in MotoMaxis/DS203-DSOQuad taking up all slots?

### ISSUE THREAD

**Issue by mofosyne** [jelinj8/dso203#4]
https://github.com/MotoMaxis/DS203-DSOQuad

I noticed that installing https://github.com/MotoMaxis/DS203-DSOQuad/blob/master/WildcatV6.5/app1.hex removes access to other app in other slots.

**Comment by MotoMaxis** [comment]
Yes, this firmware takes up 2 slots. You can try loading to 3rd and 4th slot.   Also, if you have hw 2.81+ and use the custom FPGA code, then the other programs most likely won't work.

I have hw 2.82 and did install PAWN PAWN_002.HEX to 3rd slot. It loads fine but modules don't work.

From User guide:

> HARDWARE AND COMPATIBILITY ISSUES
COMPATIBILITY >> MAKE SURE THE CORRECT VERSION IS USED FOR THE HARDWARE<<
VERSIONS 6.5 is COMPATIBLE WITH:
Hardware V2.81 and later ONLY (WARNING: LOADING V6.5 ON PRE-HW 2.81 DEVICES WILL OVERWRITE THE FPGA CODE!)
Sys V1.64
FPGA 2.81 or W1.0 - 1.1 (FPGA W1.1 update is needed for full speed oversampling mode to work)
VERSION 5.6 IS COMPATIBLE WITH:
Hardware V2.6 to V2.72 but NOT including V2.81 (WARNING: LOADING V5.6 ON V2.81 AND LATER DEVICES WILL FAIL, OR POSSIBLY OVERWRITE
PART OF THE SYSTEM. There appears to be some safeguard against this, but possibly not on all DFU versions)
Sys V 1.51 and later
FPGA V2.61
ALTERBIOS V 0.4 Needs to be installed on 2Mb devices to fix file corruption issues. Developed by Petteri Aimonen, available at
http://koti.kapsi.fi/jpa/dsoquad/ (along with a variety of other interesting software for the Quad)
VERSIONS 6.5 AND 5.6 leave both “slot 3” and “slot 4” available for other programs to use. Previous versions only left slot 4 
open.
<

**Comment by mofosyne** [comment]
I see. FYI, they said that my hardware version of v2.82 is essentially the same as v2.81, but with a layout change. Whatever that means. (mine is an 8mb drive as well)

Fair enough. I wonder if there is yet a central effort to document the dso203 yet. Because dsomini.com forum is pretty disorganised. There is a seedstudio version, but it has quite a lot of dead links...

What I'm thinking would be important would be to have a central repo containing the schematics.

And perhaps maybe a header file (or json file), that describes the memory structure consistently. This would perhaps allow for some validation etc...

If none exist, I'll create such a repo.


-----

Btw there is plenty of reviews describing the dso203 as just a toy. I'm not aiming for much, but was hoping that this would still be useful for at least diagnosing serial uart lines etc... without having to crack out the big guns. What's your experience with your dso?

**Comment by MotoMaxis** [comment]
The minidso.com is the manufacturer.  They have their own firmware that was terrible and gotten better some.  Lots of reviews are based on OE firmware.  Few show real capabilities of Wildcat's firmware.  I just know the basics of it.

The start of Wildcat firmware is found here http://www.seeedstudio.com/forum/viewtopic.php?f=26&t=2957&start=690  It continued at the new seeed.cc site but it seems the thread been removed now.  I am happy jelinj8 put the firmware up on github.

There is also a wiki http://wiki.seeedstudio.com/DSO_Quad/ with device info.

I don't use my pocket dso much but I can say from experience its more than enough for automotive repairs, boat motor repairs, repairing household electronics and in RC field.  Just using it to look at how stable the voltage is, makes it valuable tool to have.  I've repaired many thing with the help of this little guy.  It does have a limited bandwidth but anything under 4Mhz, its shows nice clean square waveform if I remember correctly.  I was able to look at a 12Mhz signal once and it was ugly but the Meters were spot on.  So it has served me well for those rare times. Btw, I did get a 1:20 attenuator, hv capacitive probe and a clamp amp probe to be able to use the dso with my troubleshooting repairs, 

It always great to meet new users of this dso.  Go ahead with the repo as I am not aware of any other.
